packages/adblocker/src/compression.ts

```diff
@@ -16,12 +17,38 @@ import networkRedirectCodebook from './codebooks/network-redirect.js';
 import networkRawCodebook from './codebooks/raw-network.js';
 import cosmeticRawCodebook from './codebooks/raw-cosmetic.js';
 
+const TEXT_ENCODER = new TextEncoder();
+
+export class Smaz {
+  private compressor: SmazCompress;
+  private decompressor: SmazDecompressRaw;
+
+  constructor(codebook: readonly string[], maxSize?: number) {
+    this.compressor = new SmazCompress(codebook, maxSize);
+    this.decompressor = SmazDecompressRaw.fromStringCodebook(codebook);
+  }
+
+  public compress(buffer: string) {
+    return this.compressor.compress(TEXT_ENCODER.encode(buffer));
+  }
+
+  public getCompressedSize(buffer: string) {
+    return this.compressor.getCompressedSize(TEXT_ENCODER.encode(buffer));
+  }
+
+  public decompress(buffer: Uint8Array): string {
+    return new TextDecoder('utf8', { ignoreBOM: true }).decode(
+      this.decompressor.decompress(buffer),
+    );
+  }
+}
```

Comment:
Why do we need to encode all? I thought unicode was only allowed for a few specific types of inputs (e.g. raw network filter but not network CSP, etc.)

Reply:
Did you measure the overhead of encoding for all inputs rather than only the necessary ones?